tests/scripts/generate_psa_tests.py

```diff
@@ -608,6 +617,7 @@ def make_test_case(self, key: StorageTestData) -> test_case.TestCase:
         )
         dependencies = finish_family_dependencies(dependencies, key.bits)
         dependencies += generate_key_dependencies(key.description)
+        dependencies = fix_key_pair_dependencies(dependencies, 'BASIC')
```

Comment:
Why do we need this like? Shouldn't `generate_key_dependencies(key.description)` already have `xxx_GENERATE` which implies `xxx_BASIC`?

Reply:
I tried so, but if I modify `DEPENDENCY_FROM_KEY` as follows:
```
DEPENDENCY_FROM_KEY = {
    AES_128BIT_ONLY_DEP_REGEX: AES_128BIT_ONLY_DEP,
    r'ECC_PAIR': ["PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE"]
```
then I will have both `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE` (expected), but also `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR` which is the legacy one that we don't want.
So I think that if we keep this solution then I'll have to search&remove `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR` from the dependency list. But I cannot get the improvement in this case :/

Reply:
Yes, but I'd expect `generate_key_dependencies` to do the rewrite, not just add `xxx_GENERATE`. Which means it would need to take `dependencies` as a parameter, but that's ok, this function isn't used anywhere else (yet).

Let's keep this as is for now. I expect I'll do some refactoring soon anyway (because I'm going to need dependency-calculating functions in other work), I can take care of this.